src/measurement/store.ts

```diff
@@ -38,11 +39,15 @@ export class MeasurementStore {
 			this.redis.set(getMeasurementKey(id, 'probes_awaiting'), probes.length, { EX: probesAwaitingTtl }),
 			this.redis.json.set(key, '$', {
 				id,
-				type,
+				type: request.type,
 				status: 'in-progress',
 				createdAt: startTime.toISOString(),
 				updatedAt: startTime.toISOString(),
+				target: request.target,
+				...(request.locations.some(l => l.limit) ? {} : { limit: request.limit }),
 				probesCount: probes.length,
+				...(request.locations.length ? { locations: request.locations } : {}),
+				...(_.isEmpty(request.measurementOptions) ? {} : { measurementOptions: request.measurementOptions }),
```

Comment:
Looks like this doesn't work either because the request includes fields added by Joi at this point. We'll again have to check for non-default values.